scripts/fabtests/ccm_runfabtests.sh

```diff
@@ -0,0 +1,196 @@
+#     conditions are met:
+#
+#      - Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      - Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials
+#        provided with the distribution.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AWV
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+# BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+debug=0
+load_ccm_module="module load ccm"
+ccm_login="ccmlogin"
+ccm_login_script="${HOME}/ccm_login_fabtests_script.$$"
+fabtests_directory="`pwd`"
+fabtests_cmd="./runfabtests.sh -p "
+provider="gni"
+test_suite=""
+use_parameters=0
+verbose=""
+
+usage() {
+    ec=$1
+    echo "cray_ccm_runfabtests.sh [-d <directory>] [-D] [-p <provider>] [-t <test_suite>] [-u] [-v] [-vv] [-vvv]"
+    echo "   -d libfabric tests location"
+    echo "   -D print debug output"
+    echo "   -p libfabric provider to use"
+    echo "   -t libfabric test suite to execute"
+    echo "   -u use parameters on runfabtests"
+    echo "   -v verbose mode for runfabtests"
+    exit $ec
+}
+
+if [ $# -gt 0 ] ; then
+  while getopts "d:Dhp:t:uv:" option; do
+    case $option in
+      d) fabtests_directory=$OPTARG;;
+      D) debug=1;;
+      h) usage 0 ;;
+      p) provider="$OPTARG";;
+      t) test_suite="-t $OPTARG";;
+      u) use_parameters=1;;
+      v) verbose="${verbose} -v";;
+      *) usage 1;;
+    esac
+  done
+fi
+
+#
+# Check for ccmlogin
+#
+ccmlogin=`command -v ccmlogin`
+if [ $? != 0 ]; then
+    ${load_ccm_module}
+    if [ $? != 0 ]; then
+        echo "Error: CCM Module not loaded.  Can Not execute the FAB tests."
+        exit 1
+    fi
+
+    ccmlogin=`command -v ccmlogin`
+    if [ $? != 0 ]; then
+        echo "Error: CCM Login command not found.  Can Not execute the FAB tests."
+        exit 1
+    fi
+fi
+
+rm -f ${ccm_login_script}
+cat > ${ccm_login_script} <<"END_OF_CCMLOGIN_SCRIPT_1"
+#!/bin/bash
+
+END_OF_CCMLOGIN_SCRIPT_1
+
+echo "debug=${debug}" >> ${ccm_login_script}
+echo "fabtests_directory='${fabtests_directory}'" >> ${ccm_login_script}
+echo "provider='${provider}'" >> ${ccm_login_script}
+echo "test_suite='${test_suite}'" >> ${ccm_login_script}
+echo "use_parameters='${use_parameters}'" >> ${ccm_login_script}
+echo "verbose='${verbose}'" >> ${ccm_login_script}
+
+cat >> ${ccm_login_script} <<"END_OF_CCMLOGIN_SCRIPT_2"
+node_1=""
+node_2=""
+
+if [[ "${provider}" == "gni" ]]
```

Comment:
There probably was a problem in my script when I added this check.  I have since retried it and I can remove it.

I have remove the "provider == gni" check when getting the nid names.